pkg/selectors/kernel_test.go

```diff
@@ -225,13 +225,17 @@ func TestParseMatchArg(t *testing.T) {
 	expected1 := []byte{
 		0x01, 0x00, 0x00, 0x00, // Index == 1
 		0x03, 0x00, 0x00, 0x00, // operator == equal
-		18, 0x00, 0x00, 0x00, // length == 18
+		32, 0x00, 0x00, 0x00, // length == 32
```

Comment:
would be great to have tests for the maximum of the matched strings

Reply:
I'm not sure how we'd do that TBH.

Reply:
perhaps some that will test the code with all the different string sizes and corner cases?

Reply:
Yep, I could make a load of tests for string matches with lengths at the boundaries.